google/cloud/forseti/notifier/notifiers/inventory_summary.py

```diff
@@ -168,6 +169,35 @@ def _get_summary_data(self):
                 sorted(summary_data, key=lambda k: k['resource_type']))
             return summary_data
 
+    def _get_details_data(self):
+        """Get the detailed summarized inventory data.
+
+        Returns:
+            list: Summary details of sorted inventory data as a list of dicts.
+                Example: [{resource_type, count}, {}, {}, ...]
+
+        Raises:
+            NoDataError: If summary details data is not found.
+        """
+        LOGGER.debug('Getting inventory summary details data.')
+        with self.service_config.scoped_session() as session:
+            inventory_index = (
+                session.query(InventoryIndex).get(self.inventory_index_id))
+
+            details = inventory_index.get_details(session)
+            if not details:
+                LOGGER.error('No inventory summary detail data found for inventory '
+                             'index id: %s.', self.inventory_index_id)
+                raise util_errors.NoDataError
+
+            details_data = []
```

Comment:
There is a small amount of code duplication here (to generate `[{resource_type, count}, {}, {}, ...]`) and what's in `_get_summary_data()`. It would be nice to refactor it into a private helper method.
```
     details_data = transform_for_template(details)
     return details_data

or

     return transform_for_template(details)
```

And then transform_for_template(data) would be something like this:
```
def transform_for_template(data):
    template_data = []
    for key, value in data.iteritems():
        template_data.append(dict(resource_type=key, count=value))
    return sorted(template_data, key=lambda k: k['resource_type'])
```

Reply:
Good eye - thanks Henry!